reg/drone/srv/air_data_computer/_.0.1.uavcan

```diff
@@ -0,0 +1,48 @@
+# A generic air data computer (ADC) service.
+#
+# An air data computer service is generally a non-interactive publish-only service: when activated,
+# it keeps publishing its measurements at a fixed rate over several subjects until deactivated.
+# The activation/deactivation, if supported, is managed via the standard arming command service.
+# The data update rates and other parameters are controlled via the Register API using implementation-defined
+# register names.
+#
+# An ADC whose arming state is SLEEP is allowed to cease all network activity.
+#
+# An ADC whose arming state is DISARMED is recommended to behave as if its state was ARMED;
+# implementations are recommended to leverage this state to perform any necessary maintenance activities
+# such as calibration, self-heating, etc. The ADC should not report its own status as ARMED until said
+# maintenance activities have been completed. Therefore, the high-level controler (e.g., the flight
+# management unit) may delay the take-off until the ADC is ready.
+#
+# An ADC whose arming state is ARMED is required to comply with the requirements set out below.
+#
+# An armed ADC should publish the following data subjects synchronously at the same configurable rate
+# which should not be lower than the specified limit. The measurements should be low-pass filtered
+# to avoid frequency aliasing effects.
+#
+#   PUBLISHED SUBJECT       TYPE                                                NOTE
+#   calibrated airspeed     reg.drone.phy.kinematics.trans.Velocity1CovTs
+#   true airspeed           reg.drone.phy.kinematics.trans.Velocity1CovTs       optional
+#   pressure altitude data  reg.drone.phy.kinematics.trans.LinearCovTs          assume ISA; NED frame (+down, -up)
+#   static air data         reg.drone.phy.thermodynamics.PressureTempCovTs      pressure and OAT
+#
```

Comment:
We need the air temperature, otherwise this is unusable. Also, sensor temperature should be a must for all sensor types.

Reply:
The outside air temperature is included in `reg.drone.phy.thermodynamics.PressureTempCovTs`

Reply:
> The outside air temperature is included in reg.drone.phy.thermodynamics.PressureTempCovTs.

So then we go back to the situation of defining macro composed types that define the required set of fields that are required for a specific service to be usable by a drone.

> or it can be simply published on a dedicated subject

That doesn't make sense because you will receive the data at the same time. You don't want the PressureTemp to be received on a certain, and the rest of the data on others. The question here is: how do you make sure that you get all of this data with the same timestamp on a more grouped manner - I would say that we need to define a composed type that answers that.

Reply:
The challenge is to define types that are sufficiently rich to include all relevant data, and at the same time do not violate the interface segregation principle. Pressure and temperature are generally tightly coupled, hence we have a dedicated type here, which happens to be a structural subtype of just pressure (so one will be able to subscribe using `uavcan.si.sample.pressure.Scalar`).

Grouping everything into one large type that contains everything is against the design principles we're following here. The rationale was discussed at length earlier so there is probably no point to repeat that. Separation into multiple messages does not mean that one would be unable to process them atomically; this case is  addressed in the Specification:

<img alt="image" src="https://user-images.githubusercontent.com/3298404/96725244-5fad4300-13b9-11eb-8ced-13aabccc1652.png">

Reply:
@pavel-kirienko are you aware of the complexity of synchronizing samples from different sensors to be fused in a filter? Why are we pushing that complexity here?